modules/angular2/src/router/route_registry.js

```
@@ -25,16 +22,14 @@ export class RouteRegistry {
       MapWrapper.set(this._rules, parentComponent, recognizer);
     }
 
-    this._configFromComponent(component);
-
-    //TODO: support sibling components
-    var components = StringMapWrapper.create();
-    StringMapWrapper.set(components, 'default', component);
+    config = normalizeConfig(config);
 
-    var handler = StringMapWrapper.create();
-    StringMapWrapper.set(handler, 'components', components);
+    var components = StringMapWrapper.get(config, 'components');
+    StringMapWrapper.forEach(components, (component, _) => {
+      this._configFromComponent(component);
+    });
 
-    recognizer.addConfig(path, handler, alias);
+    recognizer.addConfig(config['path'], config, config['alias']);
```

Can we throw if the map doesn't have components and path. Since it is just a map we don't want to silently error when someone mistyped a key or forgot one.